### PR TITLE
Add `memo` parameter to `z_shieldcoinbase`

### PR DESCRIPTION
**Pull request by sellout**

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK 8c1101b5bf4c9d402f2caac34d3d48352e856398

File: doc/release-notes.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ RPC Changes
   is unable to construct the transaction without selecting funds from the transparent
   pool, so the impact of this change is that for such transactions, the user must specify
   `AllowFullyTransparent`.
+- The `z_shieldcoinbase` and RPC method now supports an optional memo.
```
Comment (str4d) [inline doc/release-notes.md:33]:
```suggestion
- The `z_shieldcoinbase` RPC method now supports an optional memo.
```

**Review by str4d — APPROVED** [review]
utACK 8c1101b5bf4c9d402f2caac34d3d48352e856398

File: doc/release-notes.md (hunk 1 above)
Comment (str4d) [reply doc/release-notes.md:33]: This will be fixed in #6605 to avoid re-running CI here.

**Pull request merged**